core/ResManager/plKeyCollector.h

```diff
@@ -21,7 +21,42 @@
 #include <map>
 #include <vector>
 
-typedef std::map<plLocation, std::map<short, std::vector<plKey>>> keymap_t;
+class plKeyVector
+{
+    std::vector<plKey> fKeys;
+    uint32_t fFlags;
+
+public:
+    plKeyVector() : fFlags() { }
+
+    uint32_t getFlags() const { return fFlags; }
+
+    bool checkFlag(uint32_t flag) const
+    {
+        return (fFlags & flag);
+    }
+
+    void setFlag(uint32_t flag, bool on = true)
+    {
+        if (on)
+            fFlags |= flag;
+        else
+            fFlags &= ~flag;
+    }
+
+    void setFlags(uint32_t flags) { fFlags = flags; }
+
+    std::vector<plKey>* operator ->() { return &fKeys; }
+    const std::vector<plKey>* operator ->() const { return &fKeys; }
+
+    std::vector<plKey>& operator *() { return fKeys; }
+    const std::vector<plKey>& operator *() const { return fKeys; }
```

Comment:
I might prefer to wrap the vector methods we need, rather than make this look like a pointer type (which it isn't)...